sklearn/metrics/_ranking.py

```diff
@@ -1083,7 +1083,12 @@ def roc_curve(
     # to make sure that the curve starts at (0, 0)
     tps = np.r_[0, tps]
     fps = np.r_[0, fps]
-    thresholds = np.r_[thresholds[0] + 1, thresholds]
+    # make sure to not have a thresholds exceeding 1 for what could look like a
+    # probability estimate and not a decision function
+    max_threshold = (
+        min(1, thresholds[0] + 1) if thresholds.max() <= 1 else thresholds[0] + 1
+    )
```

Comment:
No need to recompute `thresholds.max()`. Furthermore I thing the ternary operator does not help readability in this case:

```suggestion
    # _binary_clf_curve returns decreasing thresholds, hence:
    max_threshold = thresholds[0]
    if max_threshold <= 1:
         # Ensure that probability thresholds stay in the [0-1] range.
         max_threshold = min(1, max_threshold + 1)
    else:
         # Unbounded range for decision_function threshold values.
         max_threshold = max_threshold + 1
```

Reply:
Note I am not convinced either that this is the good way to handle this edge case but at least I find this version of the code more readable (and slightly more efficient).